Adapt build instructions to use new build scripts (#792)
RedisJSON/RedisJSON: docs/docs/developer.md

make pack          # build package (RAMP file)
make upload-artifacts   # copy snapshot packages to S3
  OSNICK=nick             # copy snapshots for specific OSNICK
make upload-release     # copy release packages to S3

common options for upload operations:
  STAGING=1             # copy to staging lab area (for validation)
  FORCE=1               # allow operation outside CI environment
  VERBOSE=1             # show more details
  NOP=1                 # do not copy, just print commands

make coverage      # perform coverage analysis
make show-cov      # show coverage analysis results (implies COV=1)
make upload-cov    # upload coverage analysis results to codecov.io (implies COV=1)

make docker        # build for specific Linux distribution
  OSNICK=nick        # Linux distribution to build for
  REDIS_VER=ver      # use Redis version `ver`
  TEST=1             # test after build
  PACK=1             # create packages
  ARTIFACTS=1        # copy artifacts from docker image
  PUBLISH=1          # publish (i.e. docker push) after build

make sanbox        # create container for CLang Sanitizer tests
```

## Building from source

> ```make build``` will build RedisJSON.

Reviewer: ```suggestion
Run ```make build``` to build RedisJSON.
```

Notes:

* Binary files are placed under `target/release/`, according to platform and build variant.

> * RedisJSON uses [Cargo](https://github.com/rust-lang/cargo) as its build system. ```make build``` will invoke both Cargo and the subsequent make command that's required to complete the build.

Reviewer: ```suggestion
* RedisJSON uses [Cargo](https://github.com/rust-lang/cargo) as its build system. ```make build``` will invoke both Cargo and the subsequent `make` command that's required to complete the build.
```

Use ```make clean``` to remove built artifacts. ```make clean ALL=1``` will remove the entire bin subdirectory.

## Running tests
There are several sets of unit tests:
* Rust tests, integrated in the source code, run by ```make cargo_test```.
* Python tests (enabled by RLTest), located in ```tests/pytests```, run by ```make pytest```.

> One can run all tests by invoking ```make test```.
> A single test can be run using the ```TEST``` parameter, e.g. ```make test TEST=regex```.

Reviewer: ```suggestion
You can run all tests with ```make test```.
To run only specific tests, use the ```TEST``` parameter. For example, run ```make test TEST=regex```.
```

> The module's test can be run against an "embedded" disposable Redis instance, or against an instance
> you provide to it. The "embedded" mode requires having the `redis-server` executable in your `PATH`.

Reviewer: ```suggestion
You can run the module's tests against an "embedded" disposable Redis instance or against an instance
you provide. To use the "embedded" mode, you must include the `redis-server` executable in your `PATH`.
```

make docker        # build for specific Linux distribution
  OSNICK=nick        # Linux distribution to build for
  REDIS_VER=ver      # use Redis version `ver`
  TEST=1             # test after build
  PACK=1             # create packages
  ARTIFACTS=1        # copy artifacts from docker image
  PUBLISH=1          # publish (i.e. docker push) after build

make sanbox        # create container for CLang Sanitizer tests
```

## Building from source
```make build``` will build RedisJSON.

Notes:

* Binary files are placed under `target/release/`, according to platform and build variant.

* RedisJSON uses [Cargo](https://github.com/rust-lang/cargo) as its build system. ```make build``` will invoke both Cargo and the subsequent make command that's required to complete the build.

Use ```make clean``` to remove built artifacts. ```make clean ALL=1``` will remove the entire bin subdirectory.

## Running tests
Python is required for RedisJSON's module test. Install it with `apt-get install python`. You'll also
need to have [redis-py](https://github.com/redis/redis-py) installed. The easiest way to get
it is using pip and running `pip install redis`.

> 

Reviewer 2: This can be removed.

    Notes on debugging, testing and documentation
---

# Developing RedisJSON

Developing RedisJSON involves setting up the development environment (which can be either Linux-based or macOS-based), building RedisJSON, running tests and benchmarks, and debugging both the RedisJSON module and its tests.

## Cloning the git repository
To clone the RedisJSON module and its submodules, run:
```sh
git clone --recursive https://github.com/RedisJSON/RedisJSON.git
```
## Working in an isolated environment
There are several reasons to use an isolated environment for development, like keeping your workstation clean and developing for a different Linux distribution.

You can use a virtual machine as an isolated development environment. To set one up, you can use [Vagrant](https://www.vagrantup.com) or Docker.

To set up a virtual machine with Docker:

```

> search=$(docker run -d -it -v $PWD:/build debian:bullseye bash)

Reviewer 2: search -> rejson

docker exec -it $search bash
```
Then run ```cd /build``` from within the container.

In this mode, all installations remain in the scope of the Docker container.
After you exit the container, you can either restart it with the previous ```docker exec``` command or save the state of the container to an image and resume it at a later time:

```
docker commit $search redisjson1
docker stop $search
search=$(docker run -d -it -v $PWD:/build redisjson1 bash)

> docker exec -it $search bash

Reviewer 2: replace `search` with `rejson`.

```

You can replace `debian:bullseye` with your OS of choice. If you use the same OS as your host machine, you can run the RedisJSON binary on your host after it is built.

## Installing prerequisites

To build and test RedisJSON one needs to install several packages, depending on the underlying OS. Currently, we support the Ubuntu/Debian, CentOS, Fedora, and macOS.

First, enter `RedisJSON` directory.

If you have ```gnu make``` installed, you can execute,

```

> make setup

Reviewer 2: `./sbin/setup`

```
Alternatively, invoke the following:

```
./deps/readies/bin/getpy3
./sbin/setup.py

> ```

Reviewer 2: Can be removed.

> Note that ```system-setup.py``` **will install various packages on your system** using the native package manager and pip. It will invoke `sudo` on its own, prompting for permission.

Reviewer 2: `system-setup.py` -> `./sbin/setup`
or just rephrase with "this will".

$ # use an existing local Redis instance for testing the module
$ REDIS_PORT=6379 make test
```

## Debugging

> Compile after settting the environment variable `DEBUG`, e.g. `export DEBUG=1`, to include the
> debugging information.

Reviewer: ```suggestion
To include debugging information, you need to set the `DEBUG` environment variable before you compile RedisJSON. For example, run `export DEBUG=1`.
```

> Breakpoints can be added to Python tests in a single-test mode, one can set a breakpoint by using the ```BB()``` function inside a test.

Reviewer: ```suggestion
You can add breakpoints to Python tests in single-test mode. To set a breakpoint, call the ```BB()``` function inside a test.
```
